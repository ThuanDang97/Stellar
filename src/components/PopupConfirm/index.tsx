import { memo } from 'react'
import { useTheme } from 'styled-components'

// Props Type
import { PopupConfirmProps } from '@self-types/components/PopupConfirm.props'

// Utils
import { pxToRem } from '@utils/theme'

// Components
import Modal from '@components/Modal'
import { TextStyled, ViewStyled } from '@components/styled-components'
import Button from '@components/Button'
import { TitlePopupStyled } from '@components/Title/TitleStyled'

const PopupConfirm = ({
  isOpen,
  textConfirm,
  onClose,
  handleDelete,
}: PopupConfirmProps) => {
  const theme = useTheme()

  return (
    <Modal pBottom={200} minWidth={275} isOpen={isOpen} onClose={onClose}>
      <ViewStyled
        width={theme.metrics.width.full}
        display="flex"
        justifyContent="flex-start"
      >
        <TitlePopupStyled
          fontSize={theme.typography.fontSize.lg}
          lineHeight={theme.typography.fontSize.lg}
          color={theme.colors.shark}
          pBottom={theme.metrics.dimensions.sm}
          pLeft={theme.metrics.dimensions.base}
          fontWeight={theme.typography.fontWeight.medium}
        >
          Delete Confirmation
        </TitlePopupStyled>
      </ViewStyled>
      <ViewStyled
        borderBottom={`${pxToRem(theme.metrics.borderWidth.default)} solid ${
          theme.colors.silverLight
        }`}
        borderTop={`${pxToRem(theme.metrics.borderWidth.default)} solid ${
          theme.colors.silverLight
        }`}
        padding={pxToRem(theme.metrics.dimensions.sm)}
        mLeft={theme.metrics.dimensions.xs}
      >
        <TextStyled
          as="p"
          fontSize={theme.typography.fontSize.md}
          lineHeight={theme.typography.fontSize.lg}
          color={theme.colors.shark}
        >
          {textConfirm}
        </TextStyled>
      </ViewStyled>
      <ViewStyled
        width={theme.metrics.width.full}
        display="flex"
        justifyContent="flex-end"
        gap={theme.metrics.dimensions.base}
        pTop={theme.metrics.dimensions.sm}
      >
        <Button
          bgColorHover={theme.colors.ironLight}
          onClick={onClose}
          size="medium"
          title="Cancel"
          color={theme.colors.shark}
          variant="dark"
        />
        <Button
          onClick={handleDelete}
          size="medium"
          variant="warning"
          title="Delete"
          bgColor={theme.colors.pomegranate}
          bgColorHover={theme.colors.cardinal}
        />
      </ViewStyled>
    </Modal>
  )
}

export default memo(PopupConfirm)
