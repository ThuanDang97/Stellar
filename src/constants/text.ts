export const TEXT_FORGOT_PASSWORD =
  'No worries! Enter your email and we will send you a link to reset.'

export const BANNER_TITLE = {
  NO_LINK_BILL: 'You’re Almost Done',
  LINKED_BILL: 'Woohoo! You did it.',
}

export const BANNER_DESCRIPTION = {
  NO_LINK_BILL:
    'Let’s set up your first bill and Stellar will make sure they get paid on time!',
  LINKED_BILL:
    'You got rid of the gravity holding down your credit score. Keep adding bills to boost your score.',
}

export const SUBSCRIPTION_FLEX_DESCRIPTION =
  'Bill credit by paying all your bills on your own flexible schedule.'
export const SUBSCRIPTION_LAUNCH_DESCRIPTION =
  'Autopay all your bills in one place and build your credit the easy way.'

export const CONNECT_BANK_DESCRIPTION_SUCCESS =
  'Your Account has been added. Would you like to link another account?'

export const CONNECT_BANK_DESCRIPTION =
  'Start by linking your checking account to allow us to automatically pay your bills.'

export const SUCCESS = 'Success!'

export const TITLE_CONNECT_BANK = `Let's connect your bank account.`

export const IDENTITY_DESCRIPTION =
  'Please use your legal name. Your date of birth will help us confirm your identity.'

export const DELETE_CONFIRM = {
  DELETE_GOAL: 'Are you sure you want to delete this item?',
  DELETE_LINK_BILL: 'Are you sure you want to delete this linked bill?',
  DELETE_BILL: 'Are you sure you want to delete this bill?',
}

export const DELETE_SUCCESS = {
  GOAL: 'Item has been deleted.',
  LINKED_BILL: 'The linked bill has been deleted.',
}

export const EDIT_SUCCESS = 'Your changes have been successfully saved.'

export const TITLE_CONNECT_ACCOUNT = 'Connect Account'

export const ADD_FIRST_BILL_TITLE = `Congrats, you're about to set up your first bill with Stellar!`
export const ADD_FIRST_BILL_DESC =
  'Once you link the bill, we will pay it on time and draft the funds from your bank account on the selected date.'

export const TITLE_CONFIRM_DETAIL = 'Confirm Details'

export const TITLE_EDIT_LINKED_BILL = 'Edit Linked Bill'

export const SNACKBAR_SUCCESS = {
  ADD_GOAL: 'Your new goals were successfully added.',
  ADD_BILL: 'Your bill was successfully added.',
  EDIT_BILL: 'Your changes have been successfully saved.',
  ADD_LINK_BILL: 'Your bill was successfully linked.',
  DELETE_GOAL: 'Your goal was successfully deleted.',
  DELETE_BILL: 'Your bill was successfully deleted.',
  DELETE_LINK_BILL: 'Your linked bill was successfully deleted.',
}

export const PAY_PERIOD = {
  PAY_MONTHLY: 'Reported Monthly',
  PAY_THREE_MONTHS: 'Reported 3 months',
  PAY_SIX_MONTHS: 'Reported 6 months',
  PAY_ANNUALLY: 'Reported Annually',
}
