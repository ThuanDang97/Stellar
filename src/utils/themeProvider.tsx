import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
