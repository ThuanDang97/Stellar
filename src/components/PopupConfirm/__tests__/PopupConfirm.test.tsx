import { fireEvent, screen } from '@testing-library/react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Constants
import { DELETE_CONFIRM } from '@constants/text'

// Components
import PopupConfirm from '@components/PopupConfirm'

describe('PopupConfirm Component', () => {
  const mockCloseFunct = jest.fn()
  const mockDeleteFunct = jest.fn()

  const props = {
    isOpen: true,
    onClose: mockCloseFunct,
    handleDelete: mockDeleteFunct,
  }

  test('renders PopupConfirm unchanged', () => {
    const { asFragment } = renderWithTheme(
      <PopupConfirm {...props} textConfirm={DELETE_CONFIRM.DELETE_GOAL} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('runs close function on click icon', async () => {
    renderWithTheme(
      <PopupConfirm {...props} textConfirm={DELETE_CONFIRM.DELETE_GOAL} />,
    )

    fireEvent.click(await screen.findByTestId('close-icon'))

    expect(mockCloseFunct).toHaveBeenCalled()
  })

  test('runs close function on click backdrop', async () => {
    renderWithTheme(
      <PopupConfirm {...props} textConfirm={DELETE_CONFIRM.DELETE_GOAL} />,
    )

    fireEvent.click(await screen.findByTestId('backdrop'))

    expect(mockCloseFunct).toHaveBeenCalled()
  })

  test('runs delete function on click Delete button', () => {
    renderWithTheme(
      <PopupConfirm {...props} textConfirm={DELETE_CONFIRM.DELETE_GOAL} />,
    )

    fireEvent.click(screen.getByRole('button', { name: /Delete/i }))

    expect(mockDeleteFunct).toHaveBeenCalled()
  })

  test('do not render PopupConfirm when it is not open', () => {
    renderWithTheme(<PopupConfirm {...props} isOpen={false} />)

    expect(screen.queryByTestId('modal-content')).not.toBeInTheDocument()
  })
})
