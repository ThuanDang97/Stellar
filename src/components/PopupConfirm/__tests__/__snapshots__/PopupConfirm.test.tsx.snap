// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupConfirm Component renders PopupConfirm unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ModalStyled__BackDropStyled-sc-d5vki3-1 jTMsSi fuZKtD"
    data-testid="backdrop"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 ModalStyled-sc-d5vki3-0 bcQnwn hTNOqC hmXhmG"
    >
      <div
        class="ModalStyled__ImageWrapper-sc-d5vki3-2 eSAkbC"
        data-testid="close-icon"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="close-icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 hzmwyD"
        display="flex"
        width="100%"
      >
        <span
          class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 TitleStyled__TitlePopupStyled-sc-t58pjn-2 dGUnnO cssbkN iyHIHp"
          color="#242A30"
          font-size="24"
          font-weight="600"
        >
          Delete Confirmation
        </span>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 hvIspU"
      >
        <p
          class="TextStyled-sc-9xia41-0 ckPksP"
          color="#242A30"
          font-size="20"
        >
          Are you sure you want to delete this item?
        </p>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 fgffmQ"
        display="flex"
        width="100%"
      >
        <button
          class="ButtonStyled-sc-1ys9v52-0 dXGULF"
          color="#242A30"
          data-testid="button"
        >
          Cancel
        </button>
        <button
          class="ButtonStyled-sc-1ys9v52-0 apyBj"
          data-testid="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
